src/multivariate/dirichlet.jl

```diff
@@ -191,30 +191,30 @@ function suffstats(::Type{<:Dirichlet}, P::AbstractMatrix{Float64})
     K = size(P, 1)
     n = size(P, 2)
     slogp = zeros(K)
-    for i = 1:n
-        for k = 1:K
-            @inbounds slogp[k] += log(P[k,i])
+    for i in axes(P, 2)
+        for (k, kP) in zip(1:K, axes(P,1))
+            @inbounds slogp[k] += log(P[kP,i])
         end
     end
     DirichletStats(slogp, n)
 end
 
 function suffstats(::Type{<:Dirichlet}, P::AbstractMatrix{Float64},
-                   w::AbstractArray{Float64})
+                   w::AbstractArray{Float64,1})
     K = size(P, 1)
     n = size(P, 2)
     if length(w) != n
         throw(DimensionMismatch("Inconsistent argument dimensions."))
     end
 
-    tw = 0.
+    tw = zero(Float64)
```

Comment:
This seems unnecessarily complicated and does not change the result compared with the current implementation?

Reply:
```suggestion
    tw = 0.
```